cli/cmd/compose/compose.go

```diff
@@ -45,21 +45,21 @@ func addComposeCommonFlags(f *pflag.FlagSet, opts *composeOptions) {
 	f.BoolVarP(&opts.Quiet, "quiet", "q", false, "Only display IDs")
 }
 
-func (o *composeOptions) toProjectName() (string, error) {
+func (o *composeOptions) toProjectName() (string, *types.Project, error) {
```

Comment:
Should we just return Project here, and name the function toProject() ? the signature is strange

Reply:
I'm not sure about adequate naming here, as this has to return either name or name _and_ project.
Maybe we could use `golang.org/x/functional/Either` :P

Reply:
My point was : we could return just the project and not the name (that's inside the project). And update the caller to use p.Name (that will not require `if err != nil` in this case :) ).

Reply:
sure, but doing so we don't make it clear project has been loaded and is fully set or we just have a name. Backend code would have to check something like `if project.Services == nil` to distingish those scenarios.